### PR TITLE
feat(tft): add GetTFSetupJsonOutput()

### PR DESCRIPTION
**Pull request by apeabody**

### CODE REVIEW

File: infra/blueprint-test/pkg/tft/terraform.go [hunk 1]
```diff
@@ -419,6 +419,31 @@ func (b *TFBlueprintTest) GetTFSetupStringOutput(key string) string {
 	return terraform.Output(b.t, &terraform.Options{TerraformDir: b.setupDir, Logger: b.logger, NoColor: true}, key)
 }
 
+// GetTFSetupJsonOutput returns TF setup output for a given key as gjson.Result.
+// An empty string for key can be used to return all values.
+// It fails test if given key does not output a primitive, if setupDir is not configured, or invalid JSON.
+func (b *TFBlueprintTest) GetTFSetupJsonOutput(key string) gjson.Result {
+	if v, ok := b.setupOutputOverrides[key]; ok {
+		if !gjson.Valid(v.(string)) {
+			b.t.Fatalf("Invalid JSON: %s", v)
```
Comment (bharathkkb) [inline infra/blueprint-test/pkg/tft/terraform.go:428]:
```suggestion
			b.t.Fatalf("Invalid JSON in setup output override: %s", v)
```

File: infra/blueprint-test/pkg/tft/terraform.go [hunk 2]
```diff
@@ -419,6 +419,31 @@ func (b *TFBlueprintTest) GetTFSetupStringOutput(key string) string {
 	return terraform.Output(b.t, &terraform.Options{TerraformDir: b.setupDir, Logger: b.logger, NoColor: true}, key)
 }
 
+// GetTFSetupJsonOutput returns TF setup output for a given key as gjson.Result.
+// An empty string for key can be used to return all values.
+// It fails test if given key does not output a primitive, if setupDir is not configured, or invalid JSON.
```
Comment (bharathkkb) [inline infra/blueprint-test/pkg/tft/terraform.go:424]: I may have misunderstood but couldn't output for a key be any arbitrary json that passes gjson.Parse?

**Review by bharathkkb — APPROVED** [review]
Thanks @apeabody!

File: infra/blueprint-test/pkg/tft/terraform.go (hunk 2 above)
Comment (apeabody) [reply infra/blueprint-test/pkg/tft/terraform.go:424]: Yeah - Bad carry over from `GetTFSetupStringOutput`.

**Pull request merged**